dotCMS/src/main/java/com/dotmarketing/listeners/ContextLifecycleListener.java

```diff
@@ -21,79 +21,81 @@
-        
+                .onFailure(e -> Logger.warn(ContextLifecycleListener.class,
+                        "Shutdown : " + e.getMessage()));
+
         Try.run(() -> LicenseUtil.freeLicenseOnRepo())
-                        .onFailure(e -> Logger.warn(ContextLifecycleListener.class, "Shutdown : " + e.getMessage()));
+                .onFailure(e -> Logger.warn(ContextLifecycleListener.class,
+                        "Shutdown : " + e.getMessage()));
 
-        
         Try.run(() -> CacheLocator.getCacheAdministrator().shutdown())
-                        .onFailure(e -> Logger.warn(ContextLifecycleListener.class, "Shutdown : " + e.getMessage()));
-        
-
+                .onFailure(e -> Logger.warn(ContextLifecycleListener.class,
+                        "Shutdown : " + e.getMessage()));
 
         Try.run(() -> DotConcurrentFactory.getInstance().shutdownAndDestroy())
-                        .onFailure(e -> Logger.warn(ContextLifecycleListener.class, "Shutdown : " + e.getMessage()));
+                .onFailure(e -> Logger.warn(ContextLifecycleListener.class,
+                        "Shutdown : " + e.getMessage()));
 
         Try.run(() -> ReindexThread.stopThread())
-                        .onFailure(e -> Logger.warn(ContextLifecycleListener.class, "Shutdown : " + e.getMessage()));
+                .onFailure(e -> Logger.warn(ContextLifecycleListener.class,
+                        "Shutdown : " + e.getMessage()));
 
         Logger.info(this, "Shutdown : Finished.");
 
     }
 
-	public void contextInitialized(ServletContextEvent arg0) {
+    public void contextInitialized(ServletContextEvent arg0) {
 
         ByteBuddyFactory.init();
 
-		Config.setMyApp(arg0.getServletContext());
-
+        Config.setMyApp(arg0.getServletContext());
 
         String path = null;
-		try {
+        try {
 
             String contextPath = Config.CONTEXT_PATH;
-            if ( !contextPath.endsWith( File.separator ) ) {
+            if (!contextPath.endsWith(File.separator)) {
                 contextPath += File.separator;
             }
-			File file = new File(contextPath + "WEB-INF" + File.separator + "log4j" + File.separator + "log4j2.xml");
-			path = file.toURI().toString();
+            File file = new File(contextPath + "WEB-INF" + File.separator + "log4j" + File.separator
+                    + "log4j2.xml");
+            path = file.toURI().toString();
 
         } catch (Exception e) {
-			Logger.error(this,e.getMessage(),e);
-		}
+            Logger.error(this, e.getMessage(), e);
+        }
 
-		// Do not reconfigure if using global configuration.  Remove this if we move
+        // Do not reconfigure if using global configuration.  Remove this if we move
         // a full global configuration
-        if (System.getProperty("Log4jContextSelector").equals(BasicAsyncLoggerContextSelector.class.getName()))
+        if (System.getProperty("Log4jContextSelector")
```

Comment:
also, can this be null resulting in an NPE?

Reply:
You are correct, I think I did not see this in testing as we already set the Log4jContextSelector system property in code, so ends up never null,  but this needs a proper null check.